ET-OPS-4462 Rewrite getBank to use post

ed1286 opened the pull request:
-So we don't log the sort code and account number of users in Kibana we will now send this data via JSON body.
-This is the functionality for this ticket when used with the updated version of direct-debit https://github.com/hmrc/direct-debit/pull/5/files#diff-9f9fb54052a6c953f2d7fed0864f0c79

https://jira.tools.tax.service.gov.uk/browse/OPS-4462

ed1286 committed:
-changed getBank function to be POST sending a json with the bank details so they don't get logged in Kibana

ed1286 commented on app/ssttpdirectdebit/DirectDebitController.scala:
```diff
@@ -138,6 +138,7 @@ class DirectDebitController @Inject() (
         validFormData => {
 
           for {
+            //TODO this should just be validate bank and return a boolean
```
From what I can see, validateBank (in the direct-debit service which this service calls) is never used. We already have the sort code and acc number but we call direct-debit and that checks and then returns the same data. A Boolean (or just Ok/not Ok response) being returned here would make more sense. So we use validateBank and getBank can be removed or just left unused. This all depends whether other services are using these calls though.

pawelpanasewicz replied:
Thanks for the analysis Ed! Usually we remove the dead code. This is a perfect example of such situation, so please remove it. Thanks!

pawelpanasewicz commented on app/ssttpdirectdebit/DirectDebitConnector.scala:
```diff
@@ -59,8 +59,8 @@ class DirectDebitConnector @Inject() (
-    val queryString = s"sortCode=$sortCode&accountNumber=$accountNumber"
-    httpClient.GET[Option[BankDetails]](s"$baseUrl/direct-debit/bank?$queryString")
+    val bankDetailsRequest = BankDetailsRequest(sortCode, accountNumber)
+    httpClient.POST[Option[BankDetailsRequest], Option[BankDetails]](s"$baseUrl/direct-debit/bank", Some(bankDetailsRequest))
```
Why  to use these `Option`s? Please remove them and change post to look like:

```
httpClient.POST[BankDetailsRequest, BankDetails](...)
```

ed1286 committed:
-changed getBank function to be POST sending a json with the bank details so they don't get logged in Kibana
-fixed test finally..

ed1286 committed:
-changed getBank function to be POST sending a json with the bank details so they don't get logged in Kibana
-fixed test finally..
-further refactoring think all good now

ed1286 committed:
-changed getBank function to be POST sending a json with the bank details so they don't get logged in Kibana
-fixed test finally..
-further refactoring think all good now
-few tweaks removing traces of get bank should be good to go now

pawelpanasewicz commented on app/ssttpdirectdebit/DirectDebitController.scala:
```diff
@@ -137,24 +137,19 @@ class DirectDebitController @Inject() (
-            }
-          } yield result
-
+          directDebitConnector.validateBank(validFormData.sortCode, validFormData.accountNumber.toString).flatMap { isValid =>
```
The `accountNumber` is already a `String`. Please remove `.toString` part.

ed1286 committed:
-changed getBank function to be POST sending a json with the bank details so they don't get logged in Kibana
-fixed test finally..
-further refactoring think all good now
-few tweaks removing traces of get bank should be good to go now
